controller/ca/controller.go

```diff
@@ -141,16 +141,21 @@ func (c *CertificateController) syncSecret(key string) error {
 
 	dnsName := identity.ToDNSName()
 	secretName := identity.ToSecretName()
-	certAndPrivateKey, err := c.ca.IssueEndEntityCertificate(dnsName)
+	cred, err := c.ca.GenerateEndEntityCred(dnsName)
+	if err != nil {
+		log.Errorf("Failed to issue certificate for %s", dnsName)
+		return err
+	}
+	pk, err := cred.EncodePrivateKeyP8()
 	if err != nil {
 		log.Errorf("Failed to issue certificate for %s", dnsName)
 		return err
 	}
 	secret := &v1.Secret{
 		ObjectMeta: metav1.ObjectMeta{Name: secretName},
 		Data: map[string][]byte{
-			pkgK8s.TLSCertFileName:       certAndPrivateKey.Certificate,
-			pkgK8s.TLSPrivateKeyFileName: certAndPrivateKey.PrivateKey,
+			pkgK8s.TLSCertFileName:       cred.Crt.Certificate.Raw,
```

Comment:
Similar comment here:

```suggestion
			pkgK8s.TLSCertFileName:       cred.Certificate.Raw,
```